mongomock/collection.py

```diff
@@ -506,14 +506,15 @@ def _update(self, spec, document, upsert=False, manipulate=False,
                             if not isinstance(arr, list):
                                 continue
 
+                            arr_copy = copy.deepcopy(arr)
                             if isinstance(value, dict):
-                                for idx, obj in enumerate(arr):
+                                for obj in arr_copy:
                                     if filter_applies(value, obj):
-                                        del arr[idx]
+                                        arr.remove(obj)
```

Comment:
This seems wrong... Is it always guaranteed to be the first? Remove only removes the first match in the array...